src/DynamoCore/Core/Threading/DynamoSchedulerInternals.cs

```diff
@@ -30,6 +30,16 @@ private enum EventIndex
         private readonly List<AsyncTask> taskQueue = new List<AsyncTask>();
         private readonly TimeStampGenerator generator = new TimeStampGenerator();
 
+        public IEnumerable<AsyncTask> Tasks
+        {
+            get {
+                lock (taskQueue)
+                {
+                    return taskQueue.ToList();
+                } 
+            }
```

Comment:
yeah, this lock is needed because of the other changes made in the other parts of this partial class.